var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -75,12 +75,58 @@ def install(self, spec, prefix):
             lapackblas += spec['blas'].blas_libs
 
         if '+blas' in spec or '+lapack' in spec:
+            # note that one should not use [blas_opt] and [lapack_opt], see
+            # https://github.com/numpy/numpy/commit/ffd4332262ee0295cb942c94ed124f043d801eb6
             with open('site.cfg', 'w') as f:
-                f.write('[DEFAULT]\n')
-                f.write('libraries=%s\n'    % ','.join(lapackblas.names))
-                f.write('library_dirs=%s\n' % ':'.join(lapackblas.directories))
+                # Unfortunately, numpy prefers to provide each BLAS/LAPACK
+                # differently.
+                names  = ','.join(lapackblas.names)
+                dirs   = ':'.join(lapackblas.directories)
+                # First, workout the defaults.
+                # The section title for the defaults changed in @1.10, see
+                # https://github.com/numpy/numpy/blob/master/site.cfg.example
+                if spec.satisfies('@:1.9.2'):
+                    f.write('[DEFAULT]\n')
+                else:
+                    f.write('[ALL]\n')
+                if not ('^openblas' in spec or
+                        '^mkl' in spec or
+                        '^atlas' in spec):
+                    f.write('libraries=%s\n'    % names)
+                    f.write('library_dirs=%s\n' % dirs)
+
                 if not ((platform.system() == "Darwin") and
                         (platform.mac_ver()[0] == '10.12')):
                     f.write('rpath=%s\n' % ':'.join(lapackblas.directories))
```

Comment:
@nrichart after looking at #719 AFAICT it works fine because I actually do activate `numpy` and then do `import` and run a simple test (see OP). In other words, it works as expected.